README
This is release 0.1 of Term::Size.

Term::Size is a Perl module which provides a straightforward way to get
the size of the terminal (or window) on which a script is running.

For installation instructions, see the file INSTALL.

Tim Goodwin
<tim@uunet.pipex.com>
1997-04-23


Changes
Revision history for Perl extension Term::Size.

0.210     2021-01-07 Term-Size ( TRIAL VERSION )

   - Attempt to fix build at DragonFly BSD: include <termios.h> (like for AIX) rather than
     <sys/termios.h>
   - Rewrite Makefile.PL

0.209     2018-08-21 Term-Size

   - Tweak documentation
   - Promote to a stable release

0.208     2018-08-20 Term-Size ( TRIAL VERSION )

   - use PerlIO instead of FILE* RT#38594
   - On error, Term::Size functions now return undef in scalar context, or an empty list on
     list context RT#76292

0.207     2008-08-16 Term-Size

   - the original code for retrieving terminal size via XS ioctl is back -- and this dist is
     again Unix-centric
   - use Term::Size::Any for platform independency

0.205     2008-02-27 Term-Size-Unix

   - disabled prototypes in XS code - too misleading to be useful
   - borrowed rewritten test from Term::Size::Any (which should fix most test failures)
   - NOTE: this distribution will disappear soon, merging back to the original Term-Size dist

0.204     2006-09-13 Term-Size-Unix

   - included license in Makefile.PL

0.203.2   2006-05-21 Term-Size-Unix

   - moved sources from Term-Size to this distribution
   - bug fixed: &pixels always used fd 0 rather than fileno(f)
   - added test scripts "t/00_use.t" and "t/99_pod.t"

0.203.1   2006-05-21 Term-Size

   - bug fixed: &pixels always used fd 0 rather than fileno(f)
   - experimental support for Win32 was introduced by adding a dependency on Term::Size::Win32
   - added test scripts "t/00_use.t" and "t/99_pod.t"
   - the XS code moved to distribution Term::Size::Unix - this module is now a thin skin to
     call the appropriate module Term::Size::Unix or Term::Size::Win32

0.202     2006-05-19 Term-Size

   - inserted a note to say this is an UNOFFICIAL distribution

0.201     2006-05-18 Term-Size

   - 'test.pl' rewritten with Test::More and renamed to 't/01_basic.t'
   - an initializer is now used when creating a struct winsize to avoid what looks like a bug
     on Cygwin: ioctl(., TIOCGWINSZ, .) does not set ws_xpixel and ws_ypixel fields, leaving
     them untouched. If they contained garbage, they kept the garbage
   - attempt to restore AIX compatibility according to suggestion in CPAN RT #11539 (by jydawg
     [at] xs4all.nl): "termios.h"  is not where Unix thinks it should

0.2       1997-05-13 Term-Size

   - support filehandle arguments

0.1       1997-04-23 Term-Size

   - original version


Makefile.PL

use 5;
use ExtUtils::MakeMaker;

my $EUMM_VERSION = eval $ExtUtils::MakeMaker::VERSION;

die q{OS unsupported\n} if $^O =~ /MSWin/i;

my %META = (
    'dynamic_config' => 0,
    'meta-spec'      => { version => 2 },
    'prereqs'        => {
        develop => {
            requires => {    # author tests
                'Test::Pod' => '1.18',
            }
        },
    },
    'resources' => {
        repository => {
            type => 'git',
            url  => 'git://github.com/aferreira/cpan-Term-Size.git',
            web  => 'https://github.com/aferreira/cpan-Term-Size'
        },
        bugtracker => {
            web => 'https://github.com/aferreira/cpan-Term-Size/issues',
        },
    },
);

WriteMakefile(
    NAME   => 'Term::Size',
    AUTHOR => [
        'Tim Goodwin <tim@uunet.pipex.com>',
        'Adriano Ferreira <ferreira@cpan.org>'
    ],
    VERSION_FROM  => 'Size.pm',
($] >= 5.005 ? (
    ABSTRACT_FROM => 'Size.pm',
) : ()),
($EUMM_VERSION >= 6.3001 ? (
    LICENSE       => 'perl',
) : ()),
($EUMM_VERSION >= 6.4501 ? (
    META_MERGE    => \%META,
) : ()),
($EUMM_VERSION >= 6.6303 ? (
    TEST_REQUIRES => {
        'Test::More' => 0,
    },
) : ()),
($EUMM_VERSION < 6.6303 ? (
    PREREQ_PM => {
        'Test::More' => 0,    # test
    },
) : ()),
);


Size.pm
package Term::Size;

use strict;
use vars qw(@EXPORT_OK @ISA $VERSION);

use DynaLoader ();
use Exporter ();

@ISA = qw(Exporter DynaLoader);
@EXPORT_OK = qw(chars pixels);

$VERSION = '0.210';

bootstrap Term::Size $VERSION;

1;

=encoding utf8

=head1 NAME

Term::Size - Retrieve terminal size on Unix

=head1 SYNOPSIS

    use Term::Size;

    ($columns, $rows) = Term::Size::chars *STDOUT{IO};
    ($x, $y) = Term::Size::pixels;

=head1 DESCRIPTION

L<Term::Size> is a Perl module which provides a straightforward way to
retrieve the terminal size.

Both functions take an optional filehandle argument, which defaults to
C<*STDIN{IO}>.  They both return a list of two values, which are the
current width and height, respectively, of the terminal associated with
the specified filehandle.

C<Term::Size::chars> returns the size in units of characters, whereas
C<Term::Size::pixels> uses units of pixels.

In a scalar context, both functions return the first element of the
list, that is, the terminal width.

The functions may be imported.

If you need to pass a filehandle to either of the L<Term::Size>
functions, beware that the C<*STDOUT{IO}> syntax is only supported in
Perl 5.004 and later.  If you have an earlier version of Perl, or are
interested in backwards compatibility, use C<*STDOUT> instead.

=head1 EXAMPLES

1. Refuse to run in a too narrow window.

    use Term::Size;

    die "Need 80 column screen" if Term::Size::chars *STDOUT{IO} < 80;

2. Track window size changes.

    use Term::Size 'chars';

    my $changed = 1;

    while (1) {
            local $SIG{'WINCH'} = sub { $changed = 1 };

            if ($changed) {
                    ($cols, $rows) = chars;
                    # Redraw, or whatever.
                    $changed = 0;
            }
    }

=head1 RETURN VALUES

If there is an error, both functions return C<undef>
in scalar context, or an empty list in list context.

If the terminal size information is not available, the functions
will normally return C<(0, 0)>, but this depends on your system.  On
character only terminals, C<pixels> will normally return C<(0, 0)>.

=head1 CAVEATS

L<Term::Size> only works on Unix systems, as it relies on the
C<ioctl> function to retrieve the terminal size. If you need
terminal size in Windows, see L<Term::Size::Win32>.

Before version 0.208, C<chars> and C<pixels> used to return false on error.

=head1 SEE ALSO

L<Term::Size::Any>, L<Term::Size::Perl>, L<Term::Size::ReadKey>, L<Term::Size::Win32>.

=head1 AUTHOR

Tim Goodwin, <tim@uunet.pipex.com>, 1997-04-23.

=head1 MANTAINER

Adriano Ferreira, <ferreira@cpan.org>, 2006-05-19.

=head1 COPYRIGHT AND LICENSE

Copyright (C) 1997-05-13 by Tim Goodwin.

You may redistribute them under the same terms as Perl itself.

=cut
